src/Type.zig

```diff
@@ -833,7 +833,20 @@ pub fn bitSizeof(ty: Type, comp: *Compilation) ?u64 {
 
 /// Get the alignment of a type
 pub fn alignof(ty: Type, comp: *const Compilation) u29 {
-    if (ty.requestedAlignment(comp)) |requested| return requested;
+
+    // don't return the attribute for records
+    // layout has already accounted for requested alignment
+    if (!ty.isRecord()) {
+        if (ty.requestedAlignment(comp)) |requested| {
+            // gcc does not respect alignment on enums
+            if (ty.get(.@"enum")) |ty_enum| {
```

Comment:
Enums are not considered records so this logic is unreachable here and should go somewhere else or you should use `isEnumOrRecord`.

Reply:
This is correct. There are two issues being addressed here.
1 - if it is a record, do not return the requested alignment. Return whatever layout decided is the alignment.
2 - If it is an enum, and emulating gcc do not return the requested alignment.

Reply:
My bad, I missed the `!`.

Reply:
no worries. I could break this out into different checks for different issues. It would make it clear what is happening. It would do a few more comparisons than needed.